opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -2949,12 +2955,34 @@ namespace Opm {
         }
 
         const auto qs = WellGroupHelpers::
-            getRateVector(this->well_state_, this->phase_usage_, wname);
+            getWellRateVector(this->well_state_, this->phase_usage_, wname);
 
         this->getGuideRateValues(qs, well.isInjector(), wname, grval);
 
         return grval;
     }
+    template <typename TypeTag>
+    data::GuideRateValue
+    BlackoilWellModel<TypeTag>::
+    getGuideRateInjectionGroupValues(const Group& group) const
+    {
+        auto grval = data::GuideRateValue{};
+
+        assert (this->guideRate_ != nullptr);
+
+        const auto& gname = group.name();
+        if (this->guideRate_->has(gname, Opm::Phase::GAS)) {
+            // No guiderates exist for 'gname'.
```

Comment:
Note: This comment directly contradicts the code.  I suspect you copied this from [`getGuideRateValue()`](https://github.com/OPM/opm-simulators/blob/3df75f5ab69d11e16120b824ddcc8c774591743d/opm/simulators/wells/BlackoilWellModel_impl.hpp#L2978-L2981), but that function uses the opposite criterion,
```C++
if (! this->guideRate_->has(...))
```